pages/partials/questionFooter.ejs

```diff
@@ -115,3 +115,23 @@
         $(this).addClass('disabled');
     });
 </script>
+<script>
+    const form = $('form.question-form');
+    const getForm = () => form.serialize();
+
+    // Wait for DOM (ace, file upload, etc.)
+    $(document).ready(() => {
+        // Set form state on load and submit
+        var initialForm = getForm();
+        form.submit(() => { initialForm = getForm(); });
+
+        // Check form state on unload
+        $(window).bind('beforeunload', (event) => {
+            const currentForm = getForm();
+            if(initialForm != currentForm) {
+                (event || window.event).returnValue = null;
```

Comment:
Can you throw in a comment explaining what setting `returnValue = null` does?  Are we trying to prevent some default action here?

Reply:
Done, with some minor tweaks in the process (will describe below)